pkg/controller/creds/creds.go

```diff
@@ -149,17 +150,26 @@ func toolRefsFromTools(parentTool gptscript.Tool, parentRef toolRef, tools []str
 
 func fullToolPathName(parentRef toolRef, name string) string {
 	toolName, subTool := gtypes.SplitToolRef(name)
+
+	// If this tool's path is relative to its parent.
 	if strings.HasPrefix(toolName, ".") {
```

Comment:
nit: looks like you can de-indent the entire block here by negating this predicate and returning early

```go
if !strings.HasPrefix(toolName, ".") {
    return name
}

// ...
```